### PR TITLE
Add documentation as static website using `mkdocs`

### PR DESCRIPTION
**Pull request by fpgmaas**
**PR Checklist**

- [x] A description of the changes is added to the description of this PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added or modified a feature, documentation in `docs` is updated

**Description of changes**

Currently, the documentation is not very accessible since it is only available as markdown files in the repository, as mentioned in https://github.com/unitycatalog/unitycatalog/issues/33. This PR proposes to turn the documentation into a static website with [mkdocs](https://www.mkdocs.org/).

---> Example documentation site can be found **[here](https://fpgmaas.github.io/unitycatalog/)**

---

To reproduce locally, as also mentioned in the `README`, run:

```sh
python -m venv venv
. ./venv/bin/activate 
pip install -r requirements-docs.txt
mkdocs serve
```

This PR also adds a workflow that triggers on each PR to verify that the documentation builds without errors.

We can decide later on how to deploy the documentation.

### CODE REVIEW

**Comment by tdas** [comment]
Ohh this is amazing. I was thinking of doing something with mkdocs as well. Let's me play around with it get back to you.

**Comment by fpgmaas** [comment]
> Ohh this is amazing. I was thinking of doing something with mkdocs as well. Let's me play around with it get back to you.

Cool! Feel free to let me know if you want me to make any changes, and feel free to make any changes on top of this PR as well :)

File: mkdocs.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  - Home: README.md
+  - Tutorial: tutorial.md
+  - Unity Catalog Server: server.md
+  - CLI Tool for Catalog Management: cli.md
+
+plugins:
+  - search
+
+theme:
+  name: material
```
Comment (MrPowers) [inline mkdocs.yml:19]: Yes, love seeing this! 🤩

File: .github/workflows/deploy-docs.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      - name: Set up Python
+        run: |
+          python -m venv venv
+          . ./venv/bin/activate
+          pip install -r requirements-docs.txt
+
+      - name: Deploy documentation
+        run: |
+          . ./venv/bin/activate
+          mkdocs gh-deploy --force
```
Comment (MrPowers) [inline .github/workflows/deploy-docs.yml:24]: I think we will want this to change cause I don't think we should use GitHub pages for this project.

**Comment by MrPowers** [comment]
@fpgmaas - thanks for submitting this PR!  I think this all looks good, but can we GitHub workflows to a separate PR?

I'd like to get the docs building locally first & then brainstorm the production deploy plan.

I think we will want the docs on docs.unitycatalog.io.  What do you think?

**Comment by fpgmaas** [comment]
> @fpgmaas - thanks for submitting this PR! I think this all looks good, but can we GitHub workflows to a separate PR?
> 
> I'd like to get the docs building locally first & then brainstorm the production deploy plan.
> 
> I think we will want the docs on docs.unitycatalog.io. What do you think?

@MrPowers Thanks, I removed the deployment pipeline. I guess keeping the docs-validation pipeline makes sense? If not we can always remove that one of course! 

To me docs.unitycatalog.io sounds good :) For now, I changed the `site_url` in `mkdocs.yml` to `https://docs.unitycatalog.io`, but we can always decide on that later and change it accordingly.

**Comment by MrPowers** [comment]
LGTM

**Comment by tdas** [comment]
Thank you @fpgmaas for doing this and @MrPowers for reviewing it. Merging this.

**Review by tdas — APPROVED** [review]

**Pull request merged**